cpp/src/community/k_truss_impl.cuh

```diff
@@ -0,0 +1,1164 @@
+      // repeated
+      auto last_edge_with_triangles =
+        thrust::stable_partition(handle.get_thrust_policy(),
+                                 edges_to_num_triangles,
+                                 edges_to_num_triangles + num_edges,
+                                 [] __device__(auto edge_to_num_triangles) {
+                                   return thrust::get<1>(edge_to_num_triangles) > 0;
+                                 });
+
+      auto last_edge_with_triangles_idx = thrust::distance(edges_to_num_triangles, last_edge_with_triangles);
+      // rename the above it to last_edge_with_triangles
+
+      // Note: ensure 'edges_with_triangles' and 'cur_graph_view' have the same transpose flag
+      cugraph::edge_bucket_t<vertex_t, void, true, multi_gpu, true> edges_with_triangles(handle);
+
+      cugraph::edge_property_t<decltype(cur_graph_view), bool> edge_value_output(handle,
+                                                                                 cur_graph_view);
+
+      // rename the below to edges_with_triangles
+      edges_with_triangles.insert(edgelist_srcs.begin(),
+                                  edgelist_srcs.begin() + last_edge_with_triangles_idx,
+                                  edgelist_dsts.begin());                       
+
+      cugraph::transform_e(
+        handle,
+        cur_graph_view,
+        edges_with_triangles,
+        cugraph::edge_src_dummy_property_t{}.view(),
+        cugraph::edge_dst_dummy_property_t{}.view(),
+        cugraph::edge_dummy_property_t{}.view(),
+        [] __device__(auto src, auto dst, thrust::nullopt_t, thrust::nullopt_t, thrust::nullopt_t) {
+          return true;
+        },
+        edge_value_output.mutable_view(),
+        false);
+
+      cur_graph_view.attach_edge_mask(edge_value_output.view());
+
+      // resize the 'edgelist_srcs' and 'edgelsit_dst'
+      edgelist_srcs.resize(last_edge_with_triangles_idx, handle.get_stream());
+      edgelist_dsts.resize(last_edge_with_triangles_idx, handle.get_stream());
+      num_triangles.resize(last_edge_with_triangles_idx, handle.get_stream());
+
+      num_edges = edgelist_srcs.size();
+
+      resize_dataframe_buffer(
+        incoming_vertex_pairs, num_edges, handle.get_stream());
+
+      thrust::tabulate(handle.get_thrust_policy(),
+                       get_dataframe_buffer_begin(incoming_vertex_pairs),
+                       get_dataframe_buffer_end(incoming_vertex_pairs),
+                       [edges=edges
+                       ]
+                       __device__(edge_t idx){
+                          auto edge = thrust::make_tuple(thrust::get<1>(*(edges + idx)), thrust::get<0>(*(edges + idx)));
+                          return edge;
+                       });
+      
+      thrust::sort(
+        handle.get_thrust_policy(),
+        get_dataframe_buffer_begin(incoming_vertex_pairs),
+        get_dataframe_buffer_end(incoming_vertex_pairs));  // FIXME: No need to partition
```

Comment:
Not sure why this is necessary; we should always maintain the (edge, triangle count) pairs sorted by (dst, src).